### PR TITLE
Feature/remove request headers

### PR DESCRIPTION
**Pull request by cuonglm**
## Summary
Currently, we have "set_request_headers" config, which reflects envoy
route.Route.RequestHeadersToAdd. This commit add new config
"remove_request_headers", which reflects envoy RequestHeadersToRemove.

This is also a preparation for future PRs to implement disable user
identity in request headers feature.

## Related issues
Updates #702

**Checklist**:
- [x] add related issues
- [x] updated docs
- [x] updated unit tests
- [ ] updated UPGRADING.md
- [x] ready for review

### CODE REVIEW

File: config/policy.go [hunk 1]
```diff
@@ -80,6 +80,11 @@ type Policy struct {
 	// value of any existing value of a given header key.
 	SetRequestHeaders map[string]string `mapstructure:"set_request_headers" yaml:"set_request_headers,omitempty"`
 
+	// RemoveRequestHeaders removes a collection of headers from downstream request.
```
Comment (desimone) [inline config/policy.go:83]: nit: RemoveRequestHeaders removes a collection of headers from `a` downstream request

Comment (cuonglm) [reply config/policy.go]: Fixed 👍

**Comment by desimone** [comment]
Thank you for the PR and contribution @cuonglm 👍

**Comment by cuonglm** [comment]
@desimone @calebdoxsey After this is merged, CI failed with:

```
=== RUN   TestAuthorization/direct/domains/allowed
2:54PM DBG http request method=GET url=https://httpdetails.localhost.pomerium.io/by-domain
2:54PM DBG http request method=GET url=https://authenticate.localhost.pomerium.io/.pomerium/sign_in?pomerium_expiry=1591196348&pomerium_issued=1591196048&pomerium_redirect_uri=https%3A%2F%2Fhttpdetails.localhost.pomerium.io%2Fby-domain&pomerium_signature=A_khf59fcBxO4_khtXkvu6tLUgmc2pfXrzGdQdFcyXc%3D
    TestAuthorization/direct/domains/allowed: authorization_test.go:49: 
        	Error Trace:	authorization_test.go:49
        	Error:      	Received unexpected error:
        	            	expected redirect to openid.localhost.pomerium.io: no location header found in response headers
        	Test:       	TestAuthorization/direct/domains/allowed
=== RUN   TestAuthorization/direct/domains/not_allowed
2:54PM DBG http request method=GET url=https://httpdetails.localhost.pomerium.io/by-domain
2:54PM DBG http request method=GET url=https://authenticate.localhost.pomerium.io/.pomerium/sign_in?pomerium_expiry=1591196348&pomerium_issued=1591196048&pomerium_redirect_uri=https%3A%2F%2Fhttpdetails.localhost.pomerium.io%2Fby-domain&pomerium_signature=A_khf59fcBxO4_khtXkvu6tLUgmc2pfXrzGdQdFcyXc%3D
    TestAuthorization/direct/domains/not_allowed: authorization_test.go:57: 
        	Error Trace:	authorization_test.go:57
        	Error:      	Received unexpected error:
        	            	expected redirect to openid.localhost.pomerium.io: no location header found in response headers
        	Test:       	TestAuthorization/direct/domains/not_allowed
```

The error looks un-related, is this a flaky test or know issue?

**Pull request merged**